src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-logic.cpp

```diff
@@ -571,29 +605,36 @@ PushAvStreamTransportServerLogic::HandleAllocatePushTransport(CommandHandler & h
         return std::nullopt;
     }
 
-    IngestMethodsEnum ingestMethod = commandData.transportOptions.ingestMethod;
-
-    bool isFormatSupported = false;
-
-    for (auto & supportsFormat : mSupportedFormats)
+    // here add check for valid zoneid
+    if ((transportOptions.triggerOptions.triggerType == TransportTriggerTypeEnum::kMotion) &&
+        (transportOptions.triggerOptions.motionZones.HasValue()) && (!transportOptions.triggerOptions.motionZones.Value().IsNull()))
     {
-        if ((supportsFormat.ingestMethod == ingestMethod) &&
-            (supportsFormat.containerFormat == commandData.transportOptions.containerOptions.containerType))
+
+        auto & motionZonesList = transportOptions.triggerOptions.motionZones;
+        auto iter              = motionZonesList.Value().Value().begin();
+        uint16_t zonesize      = 0;
+        auto itsz              = motionZonesList.Value().Value().begin();
+        while (itsz.Next())
         {
-            isFormatSupported = true;
+            zonesize += 1;
         }
-    }
+        bool isValidZoneSize = mDelegate->ValidateMotionZoneSize(zonesize);
```

Comment:
Why is this a delegate callback?  We know what the max allowed size is: MaxZones, right?  Why are we giving the delegate the option of using some other size (which would be a spec violation)?

Reply:
I assume it's being done in the delegate as MaxZones is not defined in this cluster.  
I suppose an alternative would be to require that ZoneManagement be constructed first (if present), and some Nullable (there are no Zones) attribute be present on this cluster that is set as part of the constructor?